src/v/cloud_io/tests/BUILD

```diff
@@ -16,3 +16,34 @@ redpanda_test_cc_library(
         "@seastar",
     ],
 )
+
+redpanda_test_cc_library(
+    name = "s3_imposter",
+    srcs = [
+        "s3_imposter.cc",
+    ],
+    hdrs = [
+        "s3_imposter.h",
+    ],
+    include_prefix = "cloud_io/tests",
+    visibility = ["//visibility:public"],
+    deps = [
```

Comment:
s3_imposter.{cc,h} didn't change, so why did implementation_deps change?

Reply:
@andrwng pointed out that some of the deps are not used

Reply:
Mm I don't think that's quite related though. I'd pointed out there are some unnecessary things in `deps`, but Noah's pointing out that the `implementation_deps` list was removed in this move.

IIUC the general guidance is that `implementation_deps` should include things that are in the cc file but not the header, and `deps` should include things required by the header.

Reply:
andrew is right. iobuf, for example, is included in .cc but not in .h so it at least looks to be a candidate for implementation_deps. similarily for other things in the deps = [] list, but i think at least retaining what was there makes sense.